### PR TITLE
Remove deprecated collector flags

### LINKED ISSUE #2738: Remove deprecated CLI flags
We've accumulated a lot of deprecated flags, it would be good to remove them. Need to verify some of these, but I think all of them are past their expiration date. 

We **should not** clean them in a single PR since they usually require code changes, better to do one PR per flag.

* [x] admin flags #2752
* [x] collector raw ports #2751
* [x] agent tags #2753
* [x] queryPortWarning & queryHOSTPortWarning #2772
* [x] cassandra flags #2789
* [x] es.max-num-spans #2482
* [x] tls flags #2790 

```
$ grep -rin '(deprecated' .
./cmd/flags/admin.go:39:	healthCheckHTTPPortWarning = "(deprecated, will be removed after 2020-03-15 or in release v1.19.0, whichever is later)"
./cmd/flags/admin.go:40:	adminHTTPPortWarning       = "(deprecated, will be removed after 2020-06-30 or in release v1.20.0, whichever is later)"
./cmd/collector/app/builder_flags.go:45:	collectorHTTPPortWarning       = "(deprecated, will be removed after 2020-06-30 or in release v1.20.0, whichever is later)"
./cmd/collector/app/builder_flags.go:46:	collectorGRPCPortWarning       = "(deprecated, will be removed after 2020-06-30 or in release v1.20.0, whichever is later)"
./cmd/collector/app/builder_flags.go:47:	collectorZipkinHTTPPortWarning = "(deprecated, will be removed after 2020-06-30 or in release v1.20.0, whichever is later)"
./cmd/agent/app/reporter/flags.go:51:		flags.String(AgentTagsDeprecated, "", "(deprecated) see --"+agentTags)
./cmd/query/app/flags.go:42:	queryPortWarning        = "(deprecated, will be removed after 2020-08-31 or in release v1.20.0, whichever is later)"
./cmd/query/app/flags.go:43:	queryHOSTPortWarning    = "(deprecated, will be removed after 2020-12-31 or in release v1.21.0, whichever is later)"
./plugin/storage/cassandra/options.go:237:		"(deprecated) Jaeger will automatically detect the version of the dependencies table")
./plugin/storage/cassandra/options.go:241:		"(deprecated) Enable (or disable) host key verification. Use "+nsConfig.namespace+".tls.skip-host-verify instead")
./plugin/storage/es/options.go:186:		"(deprecated, will be removed in release v1.21.0. Please use "+nsConfig.namespace+".max-doc-count). "+
./pkg/config/tlscfg/flags.go:54:		flags.Bool(c.Prefix+tlsEnabledOld, false, "(deprecated) see --"+c.Prefix+tlsEnabled)
./pkg/config/tlscfg/flags.go:69:		flags.Bool(c.Prefix+tlsEnabledOld, false, "(deprecated) see --"+c.Prefix+tlsEnabled)
./pkg/config/tlscfg/flags.go:74:	flags.String(c.Prefix+tlsClientCAOld, "", "(deprecated) see --"+c.Prefix+tlsClientCA)
```

### PR DESCRIPTION
**Pull request by LostLaser**
Signed-off-by: Jacob Goldsworthy <jtgolds6@gmail.com>

## Which problem is this PR solving?
-  Resolves part of  #2738 

## Short description of the changes
- Removed the deprecated ```collectorHTTPPort``` flag and the associated warning flag ```collectorHTTPPortWarning```
- Separated the hostPort formatting from ```GetAddressFromCLIOptions``` into its' own function called ```FormatHostPort```
- Wrapped retrieval of the ```CollectorHTTPHostPort``` property with a call to ```FormatHostPort```
- Added tests for the ```FormatHostPort``` function

### CODE REVIEW

File: cmd/opentelemetry/go.sum [hunk 1]
```diff
@@ -1024,8 +1024,10 @@ github.com/vektah/gqlparser v1.1.2/go.mod h1:1ycwN7Ij5njmMkPPAOaRFY4rET2Enx7IkVv
 github.com/vektra/mockery v0.0.0-20181123154057-e78b021dcbb5/go.mod h1:ppEjwdhyy7Y31EnHRDm1JkChoC7LXIJ7Ex0VYLWtZtQ=
```
Comment (yurishkuro) [inline cmd/opentelemetry/go.sum:1024]: this change does not require go.sum change, please revert this file

**Comment by yurishkuro** [comment]
editing description - this does not resolve #2738, only part of it

File: cmd/collector/app/builder_flags.go [hunk 2]
```diff
@@ -42,7 +41,6 @@ const (
 	collectorZipkinAllowedOrigins = "collector.zipkin.allowed-origins"
 	collectorZipkinAllowedHeaders = "collector.zipkin.allowed-headers"
 
-	collectorHTTPPortWarning       = "(deprecated, will be removed after 2020-06-30 or in release v1.20.0, whichever is later)"
 	collectorGRPCPortWarning       = "(deprecated, will be removed after 2020-06-30 or in release v1.20.0, whichever is later)"
 	collectorZipkinHTTPPortWarning = "(deprecated, will be removed after 2020-06-30 or in release v1.20.0, whichever is later)"
```
Comment (yurishkuro) [inline cmd/collector/app/builder_flags.go:45]: I would prefer to remove all three of these in one PR, since they all affect similar flags

**Comment by yurishkuro** [comment]
please add a "breaking change" entry to the CHANGELOG

File: cmd/collector/app/builder_flags.go (hunk 2 above)
Comment (LostLaser) [reply cmd/collector/app/builder_flags.go:45]: I was thinking the same thing. Wasn't sure if they should be split for book keeping purposes but I'll go ahead and remove the other deprecated flags in this file.

File: CHANGELOG.md [hunk 3]
```diff
@@ -6,7 +6,12 @@ Changes by Version
 
 ### Backend Changes
 
-#### Breaking changes
+#### Breaking Changes
+
+* Remove deprecated CLI flags ([#2738](https://github.com/jaegertracing/jaeger/issues/2738), [@LostLaser](https://github.com/LostLaser)):
```
Comment (yurishkuro) [inline CHANGELOG.md:11]:
```suggestion
* Remove deprecated CLI flags ([#2751](https://github.com/jaegertracing/jaeger/issues/2751), [@LostLaser](https://github.com/LostLaser)):
```

**Comment by yurishkuro** [comment]
@LostLaser thank you!

**Pull request merged**